Feat  show updated qualities

rise-erpelding opened the pull request:
## Description
The list of items that show up at http://localhost:3000 are our updated quality list of items.

## Spec

See Issue: [39](https://sparkbox.atlassian.net/browse/FSA2021-39)

## Validation
<!-- delete anything irrelevant to this PR -->

- [ ] This PR has code changes, and our linters still pass.
- [ ] This PR has new code, so new tests were added or updated, and they pass.

### To Validate

1. Make sure all PR Checks have passed.
1. Pull down all related branches.
1. Run `npm start`
2. Navigate to http://localhost:3000, verify that it shows a list of items that appears to have some updates from the items variable in item_list.js

jordanleven commented on src/gilded_rose.spec.js:
```diff
@@ -155,4 +155,21 @@ describe('generating ordered lists of items', () => {
     const testOutput = getHtmlListFromArray(list);
     expect(testOutput).toBe(`<ol><li>${JSON.stringify(sulfuras)}</li></ol>`);
   });
+  it('updates items and generates an ordered list with the updated items', () => {
```
One minor note on formatting: it looks like there are empty lines between tests throughout this file. It doesn't _really_ matter whether there are empty lines between each test or not but I'd try to make it consistent.

jordanleven commented on src/gilded_rose.spec.js:
```diff
@@ -168,7 +167,7 @@ describe('generating ordered lists of items', () => {
     let updatedItemsHTML = '<ol>';
     updatedItemsList.forEach((item) => {
-      updatedItemsHTML += `<li>${JSON.stringify(item)}</li>`
+      updatedItemsHTML += `<li>${JSON.stringify(item)}</li>`;
```
Just a quick note on curating your commits before a PR: when we talked about this, I mentioned that I don't curate my commits after opening a PR. What I _should_ have said is that I don't curate my commits after a reviewer has started looking at it (because the reason we avoid changing the commits while the PR is in review is that if we do that, we might lose track of requested changes from our reviewer).

Looking through the order of commits here, I was wondering if you pushed your changes up, opened a PR, and then saw linting failures. As long as no one has started reviewing your PR yet, feel free to squash that fix into the commit that created the error. This way, you eliminate one commit that the review doesn't have to go over and reduce noise a bit.

jordanleven commented on src/item_list.js:
```diff
@@ -18,4 +18,4 @@ export default function getHtmlListFromArray(itemArray) {
 }
 
-export const htmlListItems = getHtmlListFromArray(items);
+export const htmlListItems = getHtmlListFromArray(updateQuality(items));
```
Having multiple functions in a single line like this can be potentially confusing...I might suggest doing something like this:

```
const updatedItems = updateQuality(items);
export const htmlListItems = getHtmlListFromArray(updatedItems);
```
This function is interesting because you're taking items, putting them into a function (`updateQuality`) and _then_ passing the return value from _that_ into `getHtmlListFromArray` and returning the output of _that_. 

![image](https://user-images.githubusercontent.com/7015466/109662920-8f274580-7b39-11eb-8dcc-34967b747dc6.png)

This is piping (i.e., taking a value and putting it through an assembly line). One of the popular functional programming libraries is called Ramda, and there's a function (`pipe`) [for something like this](https://ramdajs.com/docs/#pipe). You don't need to use it here, but that's something you might find interesting. It means your function _could_ look something like this:
```
export const htmlListItems = pipe(
  updateQuality,
  getHtmlListFromArray,
  )(items);
 ```
Just something to be aware of!

jordanleven commented on src/gilded_rose.spec.js:
```diff
@@ -155,4 +156,21 @@ describe('generating ordered lists of items', () => {
     expect(testOutput).toBe(`<ol><li>${JSON.stringify(sulfuras)}</li></ol>`);
   });
+  it('updates items and generates an ordered list with the updated items', () => {
+    const itemsList = [
```
This looks good! Don't feel compelled to include _all_ of these as part of your tests in the future. Usually, when I'm doing tests like these that involve testing multiple objects or arrays, I max out that three or so. Nothing wrong with including them all here, but there's no need to test the number of passed in `items` as long as your use cases are covered.

jordanleven reviewed (commented):
Her Rise - looks great! I loved that you're using small incremental commits in your PR. It makes it really easy to go commit by commit to review your changes.

I left a few comments throughout the PR - take a look and let me know what you think!

rise-erpelding commented:
Updated item_list.js by separating `export const htmlListItems = getHtmlListFromArray(updateQuality(items));` into two lines.

Updated tests to use a shorter array instead of full items array.
